sound/soc/sof/sof-audio.h

```diff
@@ -40,6 +40,27 @@ struct snd_sof_widget;
 struct snd_sof_route;
 struct snd_sof_control;
 
+/**
+ * struct ipc_tplg_control_ops - IPC-specific ops for topology kcontrol IO
+ */
+struct ipc_tplg_control_ops {
+	bool (*volume_put)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*volume_get)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	bool (*switch_put)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*switch_get)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	bool (*enum_put)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*enum_get)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*bytes_put)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*bytes_get)(struct snd_sof_control *scontrol, struct snd_ctl_elem_value *ucontrol);
+	int (*bytes_ext_get)(struct snd_sof_control *scontrol,
+			     const unsigned int __user *binary_data, unsigned int size);
+	int (*bytes_ext_volatile_get)(struct snd_sof_control *scontrol,
+				      const unsigned int __user *binary_data, unsigned int size);
+	int (*bytes_ext_put)(struct snd_sof_control *scontrol,
+			     const unsigned int __user *binary_data, unsigned int size);
+	void (*notify)(struct snd_sof_dev *sdev, void *ipc_control_data);
```

Comment:
The `ipc_control_data` is actually the IPC notification message itself we receive from firmware, it should be named accordingly.
I would be also tempted to change the `notify` (notify what and about what?), `change_notification` perhaps? Or `ipc_notification`
This is a callback to handle notification messages about control value/data changes from the firmware.

Reply:
how about ipc_control_data -> ipc_control_message and notify -> control_update, I'd not include ipc in control_update because it is part of the IPC ops anyway

Reply:
Are you planning to update the parameter name?